### DISCUSSION TITLE
Commands are being interpreted as "repeats" after manual window adjustments

### DISCUSSION THREAD

**Discussion by slifty** [rxhanson/Rectangle#727] [General]
I often find myself moving a window / resizing it, and wanting to put it back in it's normal place with my handy Rectangle hotkey -- for instance:

1) Signal "Top Left" via Rectangle
2) Adjust the window for whatever reason (moving out of the way temporarily for instance)
3) Signal "Top Left" via Rectangle (to put it back)

Right now if f I do this, Rectangle treats that second "top left" command a "repeat" -- but it isn't really a repeat -- I actually just want it to go to the top left since that's not where it is right now.  This can cause unpredictable behavior if I have special handling of repeat commands in Rectangle set up.

My current solution is to disable repeat commands, but I'd love to use them.

One solution to this problem would be to have the command associated with a window "reset" if the window is manually adjusted; another would be to have the "repeat" version of commands only trigger if the window is in the "correct" position given the most recent command.

**Comment by rxhanson** [comment]
For some reason I was thinking that this check was in place, but I guess not. I'll take a look.

**Reply by slifty** [reply]
I figured something out -- this is only an issue if "snap on drag "is disabled"

To repeat:

1. Disable snap on drag; enable some type of repeat command (e.g. rotate monitor)
2. Use a hotkey
3. Move the window
4. Use the same hotkey

I would expect step 4 to result in the *SAME* position as step 2; and if snap on drag is enabled it does.

**Reply by rxhanson** [reply]
Ah, that makes sense. This is an odd side effect from not having enough granularity to the settings here. That snap on drag checkbox is effectively "allow Rectangle to listen to click and drag events". It exists this way to allow users to not let Rectangle interfere with certain apps that break just from having another 3rd party app listening to click/drag events (there's only a handful that I'm aware of). I think the proper path forward would be to disconnect that behavior from that checkbox and provide a hidden setting for disabling listening to click and drag events, even though that would be rather annoying for some users.

**Reply by slifty** [reply]
@rxhanson oh interestin' -- would it be useful for me to make an actual issue for this or would that be redundant?

**Reply by rxhanson** [reply]
I went ahead and added it: https://github.com/rxhanson/Rectangle/issues/815
